### ISSUE TITLE
limit_repair method alters the input array

### ISSUE THREAD

**Issue by zStupan** [NiaOrg/NiaPy#294]
**Steps to reproduce**
```python
import numpy as np
from NiaPy.task import StoppingTask

task = StoppingTask(D=2,  benchmark='sphere', nGEN=10)

x = np.array([5.5, 3.78], dtype=np.float64)
repaired_x = task.repair(x)

print(np.array_equal(x, repaired_x)) # will print 'True'

```

**Expected behavior**
I would expect the original x array to remain unchanged.

**Actual behavior**
The limit_repair function alters the input `x` array and returns it.

**System configuration**

 - OS: Kubuntu 20.10
 - Python version: 3.8.6

I believe this is undesirable behavior. Looking at `limit_repair`'s signature (accepts an input array and returns the repaired array) it makes sense that a copy of the input array would be altered and returned, and the input array would remain unchanged. 

I have found that if you use a custom function as the tasks frepair, a function that does not alter the input array, then the MTS algorithm and the HDE algorithms that use MTS search functions break (Returned best does not equal the global best stored in the task object).

Example:
 ```python
from NiaPy.task import StoppingTask
from NiaPy.algorithms.other import MultipleTrajectorySearch
from NiaPy.benchmarks import Sphere
import numpy as np

def custom_frepair(x, l, u):
    return np.clip(x, l, u)

task = StoppingTask(D=10, nGEN=40, benchmark=Sphere(), frepair=custom_frepair)
algo = MultipleTrajectorySearch()

best_x, best_f = algo.run(task)
print(best_f, task.x_f)  # task.x_f will be smaller
```

This seems to only be an issue with MTS and related algorithms. @kb2623 Can you take a look at the MTS code? I can't figure out where exactly the issue is.

**Comment by sisco0** [comment]
I totally agree with the use of `np.clip` as the method for clipping data between two boundary values.

**Comment by firefly-cpp** [comment]
Fixed in #309
